reana_commons/utils.py

```diff
@@ -206,6 +209,49 @@ def get_disk_usage(directory, summarize=False, to_human_readable_units=None):
 
     :return: List of dicts with file name and size.
     """
+
+    def _remove_upper_level_references(path):
+        """Remove upper than `./` references.
+
+        Collapse separators/up-level references avoiding references to paths
+        outside working directory.
+
+        :param path: User provided path to a file or directory.
+        :return: Returns the corresponding sanitized path.
+        """
+        return os.path.normpath("/" + path).lstrip("/")
+
+    def _is_directory(directory_path, path):
+        """Whether the given path matches a directory or not.
+
+        :param directory_path: Directory to check files from.
+        :param path: Optional wildcard pattern to use for the check.
+        :return: Full path if it is a directory, False if not.
+        """
+        secure_path = _remove_upper_level_references(path)
+        full_path = Path(directory_path, secure_path)
+        if full_path.is_dir():
+            return full_path
+        return False
+
+    def _get_files_recursive_wildcard(directory_path, path):
+        """Get file(s) fitting the wildcard from the workspace.
+
+        :param directory_path: Directory to get files from.
+        :param path: Wildcard pattern to use for the extraction.
+        :return: list of paths sorted by length.
+        """
+        secure_path = _remove_upper_level_references(path)
+        # if `secure_path` is a directory, append `/*` to get all the files inside
+        if _is_directory(directory_path, secure_path):
+            _rstrip_path = secure_path.rstrip("/")
+            secure_path = "{}/*".format(_rstrip_path)
+        posix_dir_prefix = Path(directory_path)
+        paths = list(posix_dir_prefix.glob(secure_path))
+        # sort paths by length to start with the leaves of the directory tree
+        paths.sort(key=lambda path: len(str(path)), reverse=True)
+        return paths
```

Comment:
I see that this code is mainly copied from [reana-workflow-controller utils.py](https://github.com/reanahub/reana-workflow-controller/blob/master/reana_workflow_controller/rest/utils.py#L429). I think that instead of maintaining two copies of the same code it would be better to move it to `reana-commons` properly (not as internal functions of `du`) and change `reana-workflow-controller` code to use these functions from `r-commons`